node/src/accountant/big_int_db_processor.rs

```diff
@@ -0,0 +1,1613 @@
+            Subtraction("balance", (6) as u128),
+        true);
+    }
+
+    #[test]
+    fn early_return_for_successful_insert_works_for_addition() {
+        precise_upsert_or_update_assertion(
+            "early_return_for_successful_insert_works",
+            "insert into test_table (name,balance_high_b,balance_low_b) values (:name,:balance_high_b,:balance_low_b)",
+            None,
+            None,
+            Addition("balance", (i64::MAX - 58989) as u128), false);
+    }
+
+    #[test]
+    fn early_return_for_successful_insert_works_for_subtraction() {
+        precise_upsert_or_update_assertion(
+            "early_return_for_successful_insert_works_for_subtraction",
+            "insert into test_table (name,balance_high_b,balance_low_b) values (:name,:balance_high_b,:balance_low_b)",
+            None,
+            None,
+            Subtraction("balance", 58989787), false);
+    }
+
+    #[test]
+    fn insert_fails_simple_update_succeeds_for_addition() {
+        precise_upsert_or_update_assertion(
+            "insert_fails_simple_update_succeeds_for_addition",
+            "insert into test_table (name, balance_high_b, balance_low_b) values (:name, :balance_high_b, :balance_low_b) on conflict (name) do",
+            Some(|table_name |format!("update {} set balance_high_b = balance_high_b + :balance_high_b, balance_low_b = balance_low_b + :balance_low_b where name = :name",table_name)),
+            Some((&"Joe", 60, 5555)),
+            Addition("balance", 78985269), false);
+    }
+
+    #[test]
+    fn insert_fails_simple_update_succeeds_for_subtraction() {
+        precise_upsert_or_update_assertion(
+            "insert_fails_simple_update_succeeds_for_subtraction",
+            "insert into test_table (name, balance_high_b, balance_low_b) values (:name, :balance_high_b, :balance_low_b) on conflict (name) do",
+            Some(|table_name |format!("update {} set balance_high_b = balance_high_b + :balance_high_b, balance_low_b = balance_low_b + :balance_low_b where name = :name",table_name)),
+            Some((&"Joe", 89489612, 7841212)),
+            Subtraction("balance", 789858784879), false);
+    }
+
+    #[test]
+    fn insert_fails_update_succeeds_with_overflow_for_addition() {
+        let enough_big_number = BigIntDivider::reconstitute(48795846111, i64::MAX - 48484);
+        precise_upsert_or_update_assertion(
+            "insert_fails_update_succeeds_with_overflow_for_addition",
+            "insert into test_table (name, balance_high_b, balance_low_b) values (:name, :balance_high_b, :balance_low_b) on conflict (name) do",
+            Some(|table_name |format!("update {} set balance_high_b = balance_high_b + :balance_high_b, balance_low_b = balance_low_b + :balance_low_b where name = :name",table_name)),
+            Some((&"Joe", 60, 55554598)),
+            Addition("balance", enough_big_number as u128), true);
+    }
+
+    #[test]
+    fn insert_fails_update_succeeds_with_overflow_for_subtraction() {
```

Comment:
I think this test could be clearer. It looks like you're subtracting a big negative from a small negative, but there's also an `.abs()` and an `as u128` in there, and it's hard to follow the signs. Would subtracting a large positive number from a small positive number work just as well? It'd be easier to understand.